CONTRIBUTING.md

```diff
@@ -80,6 +101,67 @@ When two or more approaches must be compared, please write a benchmark in the be
+Component owners MUST do their best to maintain a high level of quality, security, performance, and specification compliance within their components.
+Maintainers may override the decisions of component owners, but should only do so when they feel one or more of these traits is compromised.
+
+## Component Lifecycle
+
+This repository contains many components in various stages of the component lifecycle.
+A component may be **unreleased**, **experimental**, **beta**, **stable**, **unmaintained**, or **deprecated**; see the below definitions for each stability level.
+With the exception of the stable status, it is up to each individual [component owner](#component-ownership) to determine the status of a component.
+A component may only be marked stable with the approval of a member of @open-telemetry/javascript-maintainers; see the definition of stable below for more details.
+
+### Unreleased
+
+Unreleased components are in active development and have not yet been released to NPM.
+Unreleased packages should have the property `private` in their `package.json` set to `true`.
+
+### Experimental
+
+Experimental packages are in active development.
+They should be considered unstable and potentially unsuitable for production use.
+They are released to NPM for developers and early adopters.
+Experimental components MUST have their major version set to `0`.
+If a component does not have an explicit status in its README.md file, it should be considered to be experimental.
+
+### Beta
+
+Beta packages are not yet considered stable, but an effort should be made to preserve stability across versions if possible.
+They may be ready for production use, but users should understand that their APIs or the telemetry they output MAY change if required.
+Beta components MUST have their major version set to `0`.
+
+### Stable
+
+This is the highest level of quality and maintainership guarantee provided in this repository.
+Stable packages should be considered stable and ready for production use.
+In order for a package to be marked stable, it must meet the following requirements:
+
+- It MUST have a component owner that the JS maintainers feel confident will be responsive to issues and pull requests and will fulfill their responsibility competently.
+  If a component owner is not responsive to issues and PRs, the maintainers may assign a new owner or change the status of the component to unmaintained.
+- All relevant specification relating to the component MUST be stable. For example, telemetry emitted by an instrumentation library should rely on a stable semantic convention.
+- It MUST be reviewed and approved by a member of @open-telemetry/javascript-maintainers.
+
+Beta components MUST have their major version set to `1` or greater.
+
+### Unmaintained
+
+A component which does not have an assigned component owner, or has a component owner who has been unresponsive to issues and pull requests may be marked as unmaintained.
```

Comment:
Should we mention a specific time that it takes for a `(unreleased|experimental|beta|stable) -> unmaintained` transition to happen? :thinking:

Reply:
We can do that in the future but until we have tooling to enforce it I think we don't want to make specific promises about timelines. Otherwise people will be upset when we don't adhere to them strictly.

Reply:
Makes sense. Thanks :slightly_smiling_face: